src/main/java/com/linkedin/kmf/services/ClusterTopicManipulationService.java

```diff
@@ -0,0 +1,248 @@
+      } catch (Exception e) {
+        LOGGER.error("{} {} failed to run createDeleteClusterTopic()", _configDefinedServiceName,
+            ClusterTopicManipulationService.this.getClass().getSimpleName(), e);
+      }
+    }
+  }
+
+  private void createDeleteClusterTopic() {
+
+    if (_isOngoingTopicCreationDone) {
+
+      int random = ThreadLocalRandom.current().nextInt();
+      _currentlyOngoingTopic = XinfraMonitorConstants.TOPIC_MANIPULATION_SERVICE_TOPIC + Math.abs(random);
+
+      try {
+        int brokerCount = _adminClient.describeCluster().nodes().get().size();
+        _createTopicsResult = _adminClient.createTopics(Collections.singleton(
+            new NewTopic(_currentlyOngoingTopic, XinfraMonitorConstants.TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS,
+                (short) brokerCount)));
+
+        _totalPartitions.set(brokerCount * XinfraMonitorConstants.TOPIC_MANIPULATION_TOPIC_NUM_PARTITIONS);
+
+        _isOngoingTopicCreationDone = false;
+        LOGGER.trace("Initiated a new topic creation. topic information - topic: {}, cluster broker count: {}",
+            _currentlyOngoingTopic, brokerCount);
+      } catch (InterruptedException | ExecutionException e) {
+        LOGGER.error("Exception occurred while retrieving the brokers count: ", e);
+      }
+    }
+
+    try {
+      LOGGER.trace("cluster id: {}", _adminClient.describeCluster().clusterId().get());
+      Collection<Node> brokers = _adminClient.describeCluster().nodes().get();
+      if (_createTopicsResult.all().isDone() && doesClusterContainTopic(_currentlyOngoingTopic, brokers,
+          _adminClient)) {
+        _adminClient.deleteTopics(Collections.singleton(_currentlyOngoingTopic)).all();
+        LOGGER.trace("clusterTopicManipulationServiceRunnable: Initiated topic deletion on {}.",
+            _currentlyOngoingTopic);
+        _isOngoingTopicCreationDone = true;
+        LOGGER.trace("{}-clusterTopicManipulationServiceRunnable successful!", this.getClass().getSimpleName());
+      }
+    } catch (InterruptedException | ExecutionException e) {
+      LOGGER.error("Exception occurred while creating cluster topic in {}: ", _configDefinedServiceName, e);
+    }
+  }
+
+  private boolean doesClusterContainTopic(String topic, Collection<Node> brokers, AdminClient adminClient)
+      throws ExecutionException, InterruptedException {
+
+    for (Node broker : brokers) {
+      LOGGER.trace("broker log directories: {}",
+          adminClient.describeLogDirs(Collections.singleton(broker.id())).all().get());
+      Map<Integer, Map<String, DescribeLogDirsResponse.LogDirInfo>> logDirectoriesResponseMap =
+          adminClient.describeLogDirs(Collections.singleton(broker.id())).all().get();
+
+      this.processBroker(logDirectoriesResponseMap, broker, topic);
+    }
+    return _totalPartitions.get() == 0;
+  }
+
+  void processBroker(Map<Integer, Map<String, DescribeLogDirsResponse.LogDirInfo>> logDirectoriesResponseMap,
```

Comment:
Two things.

1. Please add Java doc to describe the purpose of this method and how it achieves the purpose. It took me a while to understand that it's doing

2. Why is the `_totalPartitions ` needed? Can we just have `processBroker(...)` return the number of partitions it has discovered and sum them up. Then compare it with the expected total number of partitions?

Reply:
JAVA doc was appended.